### DISCUSSION TITLE
Danfoss 014g2461 partially loses connection after HA restart

### DISCUSSION THREAD

**Discussion by laurencijs** [Koenkk/zigbee2mqtt#14773] [Devices]
Hello!

I’m not sure if this is an Z2M or HomeAassistant issue, but anyway somewhere needs to start :)

In short, I have a pretty simple setup, 3 Danfoss 014g2461, connected to Zigbee2MQTT and HA on top.
Temperature set points are done manually or semi-manually, with no weird setups, just automation triggering changes.
Coordinator: CC2652P USB Dongle

The issue is, that after the HA update followed by the HA restart I lose connection to the random valve, but almost definitely to at least one.
To be more specific - it can read the current temperature and R/O parameters, but any update is not working including “occupied_heating_setpoint” which is essential for my setup. Tried from HA and z2m interface - not working in both.
To get it back to normal I need to go through the factory reset procedure:
-) force remove the device from z2m
-) stop z2m
-) f. reset valve
-) start z2m
-) connect valve
-) rename it back to the previous name

Quite annoying to do that after every update. 😄 
Any ideas about what is wrong and how can I get it to work normally?

**Comment by paszczaq** [comment]
Hi!
I'm facing similar issue to yours and basically same as described as an bug: https://github.com/Koenkk/zigbee2mqtt/issues/13478
In my case only 1 Danfoss Ally Valve is loosing connection with coordinator (Sonoff Zigbee USB Dongle based on CC2652) and its loosing connection only from coordinator side so I can't send any changes to Danfoss Ally but valve sends upates to coordinator.

Check if you have sam issue about LQI on map - after it stop respon
![termostaty_mapa](https://user-images.githubusercontent.com/2835696/199690338-6d5108f0-0743-49dd-82b5-204973ff41ec.png)
ding to commands the LQI is 0

Its really annoying becose those TRV's works pretty well.

**Comment by paszczaq** [comment]
Just to add - to recover TRV it should be sufficient to:
1. shutdown Z2M
2. powercycle TRV by pulling out battery
3. start z2m

**Comment by drblah** [comment]
Hi I have this issue too, but I have found, at least part of the issue.

My home network consists of 3 Ikea Trådfri and 3 Danfoss TRV. If I power cycle the TRV they are recovered as @paszczaq mentions. But after 24-48 hours, the TRV will stop responding to commands from Zigbee. However, I noticed that if I never turn off power to my Ikea lights, some of the TRV will start using the lights as routers. Then they will keep working until the lights are powered down.

Now, This is only a partial solution. It only works for 2 of my TRV which are very close to Ikea lights (within 2 meters). My third TRV which is more than 4 meters away will stay directly connected to my CC2652P dongle and eventually stop responding to commands.

To me, it appears that as if the TRV has very poor connection handling. They appear unable to re-connect if the Zigbee network changes or goes down temporarily. Perhaps this is something we should report directly to Danfoss? Maybe they could push an update?

**Reply by lux73** [reply]
a have experienced this with one of my four Danfoss too (two newly purchased - one of the new have Problems)

**after upgrading my CC2562RB Stick to the latest dev Firmware 20221226 & updating my z2m to 1.29.2 all problems are gone**

**Comment by laurencijs** [comment]
Hi guys,

at first, I didn't notice as usual - you don't feel something if everything is alright :D but it has been working flawlessly already for some time. Perhaps it was after some update, as I didn't notice it I can't even tell if it was a z2m or HA update, but for now, everything works fine. Have already restarted many times - no problems at all. I hope it will stay that way :D

@drblah I don't have any Ikea devices in my mesh, as routers I use only Aqara labeled  [Xiaomi SP-EUC01](https://www.zigbee2mqtt.io/devices/SP-EUC01.html#xiaomi-sp-euc01) I'm considering smart light switches as router/relay as it is always powered and does not need extra space, but haven't found yet any satisfying device, I have a bad experience with "no neutral wire" smart switches - light is flickering and this is getting annoying and I do have a neutral wire in all places I need. I'm using only LED bulbs. If the switch will have a router inbuilt it will require even more power, so the possibility of flickering is even higher. So I'm still looking for something suitable for me. 
If it is possible you can try to restart TRVs with all lights powered off, maybe it will reach the coordinator directly, but I'm not sure if the signal will be too weak or if it will reconnect when the relay will appear or not, sorry I'm not very strong in how Zigbee works.

**Comment by laurencijs** [comment]
@lux73 Yes firmware upgrades can change many things. Unfortunately, I don't see any newer than 20220219 for CC2652P. 

The bad news for me is that issue appeared again, I don't know what has changed, I didn't have a lost connection issue for several months, but now it happens from time to time again.

**Reply by lux73** [reply]
the mentioned Firmware ist in the development Tree

afaik it should be marked as stable soon
